### PR TITLE
Use Vite for Storybook

### PR DESCRIPTION
**Pull request by jandrade**
## Summary:

After we switched to SB7, I noticed that I broke the ability to watch
for WB file changes (via Webpack aliases).

I'm fixing this and taking the opportunity to switch to Vite to improve
start times and overall DevEx.

By doing this change, Hot Module Reloading will be faster and more
reliable.

Also made the following changes:

- Updated all the Storybook packages to the latest versions (v7.2.0).
- Fixed an issue importing a CSS file in the preview mode.

### Benchmarks

```
Vite cold start time:      5.63 s for manager and 10 s for preview 
Vite hot start time:       5.29 s for manager and 9.39 s for preview 

Webpack cold start time:   5.88 s for manager and 29 s for preview 
Webpack hot start time:    5.45 s for manager and 23 s for preview
````

Issue: XXX-XXXX

## Test plan:

Verify in Chromatic that Storybook is working as expected.

Locally, run `yarn start` and verify that the app is working as expected, and
that HMR is working.

https://github.com/Khan/wonder-blocks/assets/843075/1f365e67-5747-4373-99f4-b6e4c670c6b5

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,17 +49,16 @@
     "@khanacademy/eslint-config": "^2.0.0",
     "@khanacademy/eslint-plugin": "^2.0.0",
     "@khanacademy/wonder-stuff-testing": "^2.4.2",
-    "@pmmmwh/react-refresh-webpack-plugin": "^0.5.10",
```
Comment (jandrade) [inline package.json:52]: **note:** 👋

File: .storybook/preview-head.html [hunk 2]
```diff
@@ -4,6 +4,7 @@
     href="https://fonts.googleapis.com/css?family=Inconsolata"
 />
 <link href="https://fonts.googleapis.com/css2?family=Source+Serif+Pro&display=swap" rel="stylesheet">
+<link rel="stylesheet" href="./sb-styles/vars.css" />
```
Comment (jandrade) [inline .storybook/preview-head.html:7]: **note:** I fixed this import that was not working properly in the Storybook instance.

Comment (jeresig) [reply .storybook/preview-head.html:7]: Huh... that's surprising that the imports weren't working properly?

**Review by jeresig — APPROVED** [review]
Those improved preview build times are very exciting!!

File: .storybook/preview-head.html (hunk 2 above)
Comment (jandrade) [reply .storybook/preview-head.html:7]: Yeah, for some reason Storybook doesn't resolve properly when we try to use `@import` inside CSS files (seems to be a problem with relative paths).

**Pull request merged**